client/rest-high-level/src/main/java/org/elasticsearch/client/security/user/privileges/Role.java

```diff
@@ -156,14 +160,13 @@ public boolean equals(Object o) {
             && indicesPrivileges.equals(that.indicesPrivileges)
             && applicationResourcePrivileges.equals(that.applicationResourcePrivileges)
             && runAsPrivilege.equals(that.runAsPrivilege)
-            && metadata.equals(that.metadata)
-            && transientMetadata.equals(that.transientMetadata);
+            && metadata.equals(that.metadata);
```

Comment:
Hey, can you elaborate on why we don't factor in the transient metadata? This means that an enabled role is equal to a disabled one